itest/test/spectre/metrics_test.py

```diff
@@ -0,0 +1,201 @@
+
+    return metrics
+
+
+def _assert_request_timing_metrics(metrics, cache_name):
+    assert len(metrics) == 4
+    assert metrics[0].dimensions == {
+        'status': '200',
+        'metric_name': 'spectre.request_timing',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': 'backend.main',
+        'instance_name': 'itest',
+        'cache_name': cache_name,
+    }
+    assert metrics[1].dimensions == {
+        'status': '200',
+        'metric_name': 'spectre.request_timing',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': '__ALL__',
+        'instance_name': 'itest',
+        'cache_name': cache_name,
+    }
+    assert metrics[2].dimensions == {
+        'status': '200',
+        'metric_name': 'spectre.request_timing',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': 'backend.main',
+        'instance_name': 'itest',
+        'cache_name': '__ALL__',
+    }
+    assert metrics[3].dimensions == {
+        'status': '200',
+        'metric_name': 'spectre.request_timing',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': '__ALL__',
+        'instance_name': 'itest',
+        'cache_name': '__ALL__',
+    }
+
+
+def _assert_fetch_hit_rate(metrics, cache_name):
+    assert len(metrics) == 2
+    assert metrics[0].dimensions == {
+        'metric_name': 'spectre.fetch_body_and_headers',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': 'backend.main',
+        'instance_name': 'itest',
+        'cache_name': cache_name,
+        'cache_status': 'miss',
+    }
+
+    assert metrics[1].dimensions == {
+        'metric_name': 'spectre.hit_rate',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': 'backend.main',
+        'instance_name': 'itest',
+        'cache_name': cache_name,
+        'cache_status': 'miss'
+    }
+
+
+def _assert_store_metric(metric, cache_name):
+    assert metric.dimensions == {
+        'metric_name': 'spectre.store_body_and_headers',
+        'habitat': 'uswest1a',
+        'service_name': 'spectre',
+        'namespace': 'backend.main',
+        'instance_name': 'itest',
+        'cache_name': cache_name,
+    }
+
+
+def test_cache_miss(log_file):
+
+    response = get_through_spectre(
+        '/timestamp/',
+    )
+    assert response.status_code == 200
+    assert response.headers['Spectre-Cache-Status'] == 'miss'
+
+    metrics = _load_metrics(log_file)
+
+    # First 2 metrics are `spectre.fetch_body_and_headers` and `spectre.hit_rate`
+    _assert_fetch_hit_rate(metrics[0:2], 'timestamp')
+
+    # Then since it's a miss we have a `spectre.store_body_and_headers`
+    _assert_store_metric(metrics[2], 'timestamp')
+
+    # Finally the `spectre.request_timing`
+    _assert_request_timing_metrics(metrics[3:7], 'timestamp')
+    assert len(metrics) == 7
```

Comment:
Shouldn't this assertion be before the `_assert...` calls? If metrics contained less than 7 elements, wouldn't metrics[3:7] raise an exception?